packages/core/src/bag3d/core/assets/export/metadata.py

```diff
@@ -183,28 +183,39 @@ def metadata(context: AssetExecutionContext):
     # is has succeeded.
     process_step_list = []
     for asset_key in asset_keys:
-        event_record = instance.get_event_records(
+        event_record_list = instance.get_event_records(
```

Comment:
I had an issue here where the 'event_record_list' was None for the test but I wanted it to keep going even with this being None. However, we probably need to add some warning or some error handling. What do you think?

Reply:
I think it is ok to keep going and write an empty, or partially empty metadata.json if one of the upstream assets were not materialized. As you say, this helps in testing.
In a prodcution setup this could happen if somone runs the the pipeline for the first time and one of the assets were never materialized. However, that would show up as a failed run in dagester for that particular asset, so that's sufficient warning imo.